### PR TITLE
ISSUE-1072: Add Apache Pinot support

### LINKED ISSUE #1072: Apache Pinot support
Add connector for Pinot and integrate Pinot metadata

### PR DESCRIPTION
**Pull request by MilanBariya**
### Describe your changes :

I worked on Adding Apache Pinot support
fix #1072 

#
### Type of change :

- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: catalog-rest-service/src/main/resources/json/schema/entity/services/connections/database/pinotdbConnection.json [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "PinotdbConnection",
+  "description": "Pinotdb Database Connection Config",
+  "type": "object",
+  "javaType": "org.openmetadata.catalog.services.connections.database.PinotdbConnection",
+  "definitions": {
+    "pinotdbType": {
+      "description": "Service type.",
+      "type": "string",
+      "enum": ["Pinotdb"],
```
Comment (ulixius9) [inline catalog-rest-service/src/main/resources/json/schema/entity/services/connections/database/pinotdbConnection.json:12]: let's follow the same naming convention that we are using for other connector,
can you please rename `Pinotdb` to `PinotDB`. (Ex: [MariaDB](https://github.com/open-metadata/OpenMetadata/blob/7f323e5ecc9670bf3fd6a1fd803abd887d066ede/catalog-rest-service/src/main/resources/json/schema/entity/services/databaseService.json#L25), [DynamoDB](https://github.com/open-metadata/OpenMetadata/blob/7f323e5ecc9670bf3fd6a1fd803abd887d066ede/catalog-rest-service/src/main/resources/json/schema/entity/services/databaseService.json#L30))

File: catalog-rest-service/src/main/resources/json/schema/entity/services/connections/database/pinotdbConnection.json [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      "default": "Pinotdb"
+    },
+    "pinotdbScheme": {
+      "description": "SQLAlchemy driver scheme options.",
+      "type": "string",
+      "enum": ["pinot"],
```
Comment (ulixius9) [inline catalog-rest-service/src/main/resources/json/schema/entity/services/connections/database/pinotdbConnection.json:18]: pinotdb also supports the schema `pinot+http` & `pinot+https`, let's add that to the enum as well

**Review by ulixius9 — CHANGES_REQUESTED** [review]
Added few review comments, also can you please add the unit test for source_connection url.

File: catalog-rest-service/src/main/resources/json/schema/entity/services/databaseService.json [hunk 3]
```diff
@@ -193,6 +197,9 @@
             },
             {
               "$ref": "./connections/database/sampleDataConnection.json"
+            },
+            {
+              "$ref": "./connections/database/pinotDBConnection.json"
```
Comment (pmbrull) [inline catalog-rest-service/src/main/resources/json/schema/entity/services/databaseService.json:202]: This reference is failing due to the filename. The filename should be `pinotDBConnection.json`.

**Review by ulixius9 — APPROVED** [review]

**Review by pmbrull — APPROVED** [review]

**Pull request merged**